### PR TITLE
SentinelOne v2: Add 2 new commands

### PR DESCRIPTION
**Pull request by chloerongier**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
This PR adds 2 commands to the integration `SentinelOne v2` :
 - **sentinelone-get-accounts** : returns information (such as the ID) about an account.
 - **sentinelone-get-threat-notes** : returns the notes of a threat.

The argument "columns" is also added to the commands **sentinelone-list-agents** and **sentinelone-get-events** in order to display additional fields in the results.

Finally, the description and type of the output "_SentinelOne.Threat.Mitigation.Action_" from the command *sentinelone-mitigate-threat* was fixed.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by ilappe** [comment]
waiting for partner approval

**Comment by kgal-pan** [comment]
@munna-metron @parag-metron @saurabh-metron - Can you please review the PR and let us know if you approve of such changes?

**Comment by munna-metron** [comment]
HI @kgal-pan - Thanks for letting us know. Will review the PR and let you know the status of the same.

**Comment by munna-metron** [comment]
HI @kgal-pan , 
Reviewed the changes in the PR and those are looking good. so we can proceed further

File: Packs/SentinelOne/Integrations/SentinelOne-V2/README.md [hunk 1]
```diff
@@ -288,7 +289,7 @@ Applies a mitigation action to a group of threats that match the specified input
 | --- | --- | --- |
 | SentinelOne.Threat.ID | String | The threat ID. | 
 | SentinelOne.Threat.Mitigated | Boolean | Whether the threat was successfully mitigated. | 
-| SentinelOne.Threat.Mitigation.Action | Number | Number of threats affected. | 
+| SentinelOne.Threat.Mitigation.Action | String | The mitigation action performed. |
```
Comment (amshamah419) [inline Packs/SentinelOne/Integrations/SentinelOne-V2/README.md:292]: @chloerongier Will this break backwards compatibility for playbooks, or is this just correcting the type to match what is currently available?

File: Packs/SentinelOne/Integrations/SentinelOne-V2/SentinelOne-V2.py [hunk 2]
```diff
@@ -1588,6 +1602,36 @@ def write_threat_note(client: Client, args: dict) -> CommandResults:
...
 
+def get_threat_notes(client: Client, args: dict) -> CommandResults:
+    """
+    Get the note of a particular threat.
+    """
+    threat_id = args.get('threat_id')
+
+    context_entries = []
+    notes = client.get_threat_notes_request(threat_id)
+    if notes:
```
Comment (amshamah419) [inline Packs/SentinelOne/Integrations/SentinelOne-V2/SentinelOne-V2.py:1613]: What do we return if there are no notes found?

File: Packs/SentinelOne/Integrations/SentinelOne-V2/SentinelOne-V2.py [hunk 3]
```diff
@@ -103,6 +103,11 @@ def get_agents_outputs(agents):
             'CreatedAt': agent.get('createdAt'),
             'SiteName': agent.get('siteName'),
         }
+
+        for c in column_to_display:
+            if c in agent:
+                entry[c] = agent.get(c)
```
Comment (amshamah419) [inline Packs/SentinelOne/Integrations/SentinelOne-V2/SentinelOne-V2.py:109]:
```suggestion
        for c in set(column_to_display).intersection(agent.keys()):
            entry[c] = agent[c]
```

In this suggestion, the iteration is done over a set as opposed to a loop and is more performative.

Another solution would be to store the `column_to_display` argument as a set instead. I'm okay with the current implementation, so it's up to you whether or not you'd like to use this approach.

File: Packs/SentinelOne/Integrations/SentinelOne-V2/SentinelOne-V2.py [hunk 4]
```diff
@@ -2760,11 +2806,16 @@ def get_events(client: Client, args: dict) -> Union[CommandResults, str]:
             'EventID': event.get('id'),
         })
 
+        for c in column_to_display:
+            if c in event:
+                contents[-1][c] = event.get(c)
```
Comment (amshamah419) [inline Packs/SentinelOne/Integrations/SentinelOne-V2/SentinelOne-V2.py:2811]:
```suggestion
        for c in set(column_to_display).intersection(event.keys()):
             contents[-1][c] = event[c]
```

Same reasoning as above, but it's up to you if you want to implement it.

File: Packs/SentinelOne/Integrations/SentinelOne-V2/SentinelOne-V2.py [hunk 5]
```diff
@@ -3044,6 +3095,39 @@ def download_fetched_file(client: Client, args: dict) -> list[CommandResults]:
...
+    """
+    Get accounts info (ID, etc).
+    """
+    account_id = args.get('account_id', None)
+
+    context_entries = []
+    # Make request and get raw response
+    accounts = client.get_accounts_request(account_id)
+
+    if accounts:
```
Comment (amshamah419) [inline Packs/SentinelOne/Integrations/SentinelOne-V2/SentinelOne-V2.py:3108]: What happens if `accounts` is just an empty dict?

File: Packs/SentinelOne/Integrations/SentinelOne-V2/SentinelOne-V2.yml [hunk 6]
```diff
@@ -510,8 +513,8 @@ script:
       description: Whether the threat was successfully mitigated.
       type: Boolean
     - contextPath: SentinelOne.Threat.Mitigation.Action
-      description: Number of threats affected.
-      type: Number
+      description: The mitigation action performed.
```
Comment (amshamah419) [inline Packs/SentinelOne/Integrations/SentinelOne-V2/SentinelOne-V2.yml:516]: See my comment in the README

**Review by amshamah419 — CHANGES_REQUESTED** [review]
Great work! Just a couple comments and a question about the context output change.

File: Packs/SentinelOne/Integrations/SentinelOne-V2/README.md (hunk 1 above)
Comment (chloerongier) [reply Packs/SentinelOne/Integrations/SentinelOne-V2/README.md:292]: Hello, This is just correcting the type to match what is currently done by the code!

File: Packs/SentinelOne/Integrations/SentinelOne-V2/SentinelOne-V2.py (hunk 2 above)
Comment (chloerongier) [reply Packs/SentinelOne/Integrations/SentinelOne-V2/SentinelOne-V2.py:1613]: If no notes are found, nothing is set in the context and the readable output is :
"
Sentinel One - Get Threat Notes
No entries.
"

File: Packs/SentinelOne/Integrations/SentinelOne-V2/SentinelOne-V2.py (hunk 3 above)
Comment (chloerongier) [reply Packs/SentinelOne/Integrations/SentinelOne-V2/SentinelOne-V2.py:109]: I will update the function with your suggestion as it is more performative ! I realised that I forgot to declare the arg "columns_to_display" as optional as it is used elsewhere without this arg so I will also fix that.

**Comment by amshamah419** [comment]
Great, thanks for the changes! 

I think we are okay to proceed. Let's just update the docker image by running `demisto-sdk format -ud -i Packs/SentinelOne/Integrations/SentinelOne-V2/SentinelOne-V2.yml` and then take a look at the lint and typing issues. Afterwards, I think we can go ahead and merge!

**Comment by chloerongier** [comment]
Hello @amshamah419, we updated the python docker image and fixed the typing issues. Is everything good to be merged now ?

**Comment by amshamah419** [comment]
@chloerongier  I think are are almost good to go. Can you take a look at this Unit Test that is failing?
```bash
[2023-11-22 17:24:10,570] [INFO] SentinelOne-V2
/home/circleci/project/Packs/SentinelOne/Integrations/SentinelOne-V2/SentinelOne-V2.py:1611:45: error:
Argument 1 to "get_threat_notes_request" of "Client" has incompatible type
"Optional[Any]"; expected "str"  [arg-type]
        notes = client.get_threat_notes_request(threat_id)
                                                ^~~~~~~~~
Found 1 error in 1 file (checked 1 source file)
```

Afterwards, I think we can merge.

**Comment by chloerongier** [comment]
Hello @amshamah419, I removed the argument's type str from the function and the test is now passed !

**Review by amshamah419 — APPROVED** [review]

**Pull request merged**